presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java

```diff
@@ -194,7 +187,7 @@ public static DistributedQueryRunner createQueryRunner(
 
             if (!metastore.getDatabase(METASTORE_CONTEXT, TPCH_BUCKETED_SCHEMA).isPresent()) {
                 metastore.createDatabase(METASTORE_CONTEXT, createDatabaseMetastoreObject(TPCH_BUCKETED_SCHEMA));
-                copyTpchTablesBucketed(queryRunner, "tpch", TINY_SCHEMA_NAME, createBucketedSession(Optional.empty()), tables);
+                copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, createBucketedSession(Optional.empty()), tables, true);
```

Comment:
Why still use `copyTpchTables`? The following two methods are very similar, there's no need to have a separate `copyTpchTables()`. Remove `copyTpchTables()` and just call `copyTables()` here, and convert the `tables` parameter to the required type.

```
public static void copyTpchTables(
            QueryRunner queryRunner,
            String sourceCatalog,
            String sourceSchema,
            Session session,
            Iterable<TpchTable<?>> tables,
            boolean bucketed)
    {
        log.info("Loading data from %s.%s...", sourceCatalog, sourceSchema);
        long startTime = System.nanoTime();

        for (TpchTable<?> table : tables) {
            copyTables(queryRunner, sourceCatalog, sourceSchema, session, table.getTableName().toLowerCase(ENGLISH), false, bucketed);
        }
        log.info("Loading from %s.%s complete in %s", sourceCatalog, sourceSchema, nanosSince(startTime).toString(SECONDS));
    }

    public static void copyTables(
            QueryRunner queryRunner,
            String sourceCatalog,
            String sourceSchema,
            Session session,
            Iterable<String> tables,
            boolean ifNotExists,
            boolean bucketed)
    {
        log.info("Loading data from %s.%s...", sourceCatalog, sourceSchema);
        long startTime = System.nanoTime();
        for (String table : tables) {
            copyTables(queryRunner, sourceCatalog, sourceSchema, session, table, ifNotExists, bucketed);
        }
        log.info("Loading from %s.%s complete in %s", sourceCatalog, sourceSchema, nanosSince(startTime).toString(SECONDS));
    }
```

Reply:
`copyTpchTables` has been invoked by many callers who passes in Iterable<TpchTable> whereas `copyTables` reads table name directly from String type input parameter. If you wanna make `copyTables` as the general entrance, all callers are required to change the way they pass in their table name instead of their parent type. I tend not to change the method parameters since this is on testing code path. However I'm open to change all callers to invoke the new method.

Reply:
You now have two copyTpchTables(). The second is not needed.

Reply:
@fgwang7w See this: 
```
public static void copyTpchTables(
            QueryRunner queryRunner,
            String sourceCatalog,
            String sourceSchema,
            Session session,
            Iterable<TpchTable<?>> tables)
    {
        copyTpchTables(queryRunner, sourceCatalog, sourceSchema, session, tables, false);
    }

    public static void copyTpchTables(
            QueryRunner queryRunner,
            String sourceCatalog,
            String sourceSchema,
            Session session,
            Iterable<TpchTable<?>> tables,
            boolean bucketed)
    {
        log.info("Loading data from %s.%s...", sourceCatalog, sourceSchema);
        long startTime = System.nanoTime();

        for (TpchTable<?> table : tables) {
            copyTables(queryRunner, sourceCatalog, sourceSchema, session, table.getTableName().toLowerCase(ENGLISH), false, bucketed);
        }
        log.info("Loading from %s.%s complete in %s", sourceCatalog, sourceSchema, nanosSince(startTime).toString(SECONDS));
    }

    public static void copyTables(
            QueryRunner queryRunner,
            String sourceCatalog,
            String sourceSchema,
            Session session,
            Iterable<String> tables,
            boolean ifNotExists,
            boolean bucketed)
    {
        log.info("Loading data from %s.%s...", sourceCatalog, sourceSchema);
        long startTime = System.nanoTime();
        for (String table : tables) {
            copyTables(queryRunner, sourceCatalog, sourceSchema, session, table, ifNotExists, bucketed);
        }
        log.info("Loading from %s.%s complete in %s", sourceCatalog, sourceSchema, nanosSince(startTime).toString(SECONDS));
    }
```
In the first copyTpchTables(), you convert the tables parameter to  Iterable<String>, then call copyTables() directly.

Reply:
The first `copyTpchTables` is used by callers who do not pass in bucket flag. If place logic to convert table parameters, then there will be duplicate code to handle the method which caller specify parameter with bucket flag.  So current call stack logic flows like this:
```
copyTpchTables(TpchTable) 
      -> copyTpchTables(TpchTable, bucketed) 
                -> copyTables(String, bucketed)
```
Are you suggesting to change to something like below code path?
```
copyTpchTables(TpchTable) 
         -> copyTables(String, bucketed).
copyTpchTables(TpchTable, bucketed) 
         -> copyTables(String, bucketed).
```

Reply:
@fgwang7w I have a demo commit here
https://github.com/prestodb/presto/pull/17260/commits/9b129333f77da09a9eeb00ab81f7b63fc5c064e9

Reply:
Thanks @yingsu00 . I made changes to both commits and make the flow more readable now. Pls help review the code again.